### ISSUE TITLE
[TODO] [Day 1-3] 병현이의 Todo list - EDA 및 F1 점수를 기준한 성능 향상 방법 기록 등

### ISSUE THREAD

**Issue by Bkankim** [AIBootcamp14/computervisioncompetition-cv-1#14]
## 📋 작업 개요
**작업 주제:** EDA & F1 점수 향상을 위한 여러 적용 기능 및 방법
**담당자:** @Bkankim 
**마감일:** 9/3 18:00  

## 📅 기간
- 시작일: 2025-09-01
- 종료일: 2025-09-03

## 📌 목표
- F1 Score 0.8+ 달성
- EDA 최신화
- 인사이트 기록 (공유용)

## ✅ Todo List
- [x] 데이터 둘러보고 약식 EDA
- [x] 백본 하이퍼 파라미터 최적화
- [x] 개선 방법론 최소 5개 탐구 & 적용
- [x] 중간 EDA 보고서 작성

## 💡 참고 사항
- 추가로 공유할 메모나 링크가 있다면 적어주세요.

### 유용한 링크
- [x] augraphy 솔루션: [링크](https://augraphy.readthedocs.io/en/latest/doc/source/augmentations/markup.html)

--- 
# 💭 CV 경진대회 — Task & Log

## 현재 성과
- 백본 하이퍼 파라미터값 최적화 -> (f1 score 0.17 -> 0.42)
- TTA 증강 (최적화 전 기본세팅) + Albumentations + ImageNet (정규화) -> (f1 score 0.42 -> 0.63)
- 5-fold (best fold) + CosineAnnealingLR 스케줄러 -> (f1 score 0.63 -> 0.74)
- 5-fold (앙상블) + 해상도 패딩 + Calibration (Temperature Scaling) (f1 score 0.74 -> 0.87)

## 추가 개선 과제 (계획)

- 모듈화 (현재 진행중) -> 코드 접목성 및 협업 목적

- RGB 변환, EXIF 회전 보정
- 백본 모델 업그레이드
- 하이퍼 파라미터 지속적 튜닝
- 백본 앙상블 (ex. ResNet50 + EfficientNet_

- VT (Vision Transformer) 적용 고려
- AutoAugment/RandAugment 적용 고려
- Pseudo Labeling 적용 고려

- Knowledge Distillation
- Self-Supervised Pre-training
- Multi-scale Feature Pyramid

## 적용 기능 (최신화)

- TTA 증강 (Multi-scale -> 다중 해상도 transform)
- K-fold 앙상블 (Stratified Sampling)
- Calibration (Temperature Scaling) - overconfident 해결 방법론
- Label smoothing
- Cutmix/Mixup
- Gradient Clipping
- Optuna 튜닝 (하드 세팅된 하이퍼 파라미터와의 충돌이 있는지 확인 必)

## EDA
- train 데이터와 test 데이터간의 편차 (도메인 시프트) 확인 (20%)
-> 테스트셋의 회전, 밝기, 블러가 성능에 핵심 변수로 작용할 가능성 매우 높음.
-> TTA 증강에 밝기, 회전, 블러 등 적용 고려
-> 개선 했음에도 갭을 충분히 메우지 못함. -> 꾸준히 테스트중
-> 추가 인사이트 : train vs test 분포 비교후 하드 검증셋 정의 必
-> 추가 인사이트2 : TTA는 간단히 4방향 점수 비교로 최적방향 1장만 선택 후 추론하도록 해야함 (TTA 사용은 必 회전 데이터가 많은 데이터셋에 유리함)

- train 데이터 절대갯수 부족 (학습 데이터 부족)
-> overfitting 가능성 매우높음.
## -> K-fold 적용 고려
-> k-fold best select 방법 해봤으나 local f1스코어와 submission f1 스코어의 갭을 확인 (0.7 <-> 0.9) -> 과적합 & 누수
-> k-fold 앙상블 (soft-voting) 적용으로 과적합 해결 테스트중
-> 추가 인사이트 : k-fold 앙상블 결과 도메인 시프트 개선 효과 확인 (20% -> 7%)

- 해상도 차이 이슈
## -> 비율 유지 패딩 방법 사용 고려
-> LongestMaxSize + PadIfNeeded로 비율 유지 및 384로 패딩 고정 방법 연구중.
-> 주의사항 : 해상도 혼용은 금지. 꼭 384로 고정하는 방법을 추구할 것. (혼용하면 캘리브레이션에 오차가 커짐)

- 백본 최적화
-> ResNet은 예전 방법론이고 152는 무거우므로 새로운 방법 검색중
-> 현재 EfficientNet-B3 으로 변경, 사용하고있으나 유의미한 결과는 얻지 못함

- 희소 클래스 문제
-> 분류 실패로 아마 여기서 F1 점수를 다 갉아먹는 것으로 생각됨.
->  class_weight (역빈도 정규화) , WeightedRandomSampler 중 하나 적용 고려중
